### ISSUE TITLE
Why is CS4014 not shown for all functions that return a task?

### ISSUE THREAD

**Issue by ramondeklein** [dotnet/roslyn#21428]
I have the following (inconsistent) behavior when using Visual Studio 2017 with C# 7 features enabled. Suppose the following code:
```
private async Task Test1Async() => await Task.Delay(1000).ConfigureAwait(false);
private Task Test2Async() => Test1Async();
```
Functionally, these functions are exactly the same but the compiler treats calling these methods different. The following code compiles, but issues a CS4014 warning:
```
private void Test() => Test1Async();   // CS4014 is shown
```
It generates the warning "_because this call is not awaited, the current method continues to run before the call is completed_". This is a proper warning, because it often indicates a flaw in your code. In case you actually want this behavior, then you can solve it by using the following code:
```
private void Test() => _ = Test1Async();   // CS4014 is not shown anymore
```
Assigning the value to `_` is a relative new feature to indicate that the value is ignored intentionally and I think it's the best way to indicate that the result should be ignored. The following code doesn't raise CS4014:
```
private void Test() => Test2Async();   // CS4014 is not shown!
```
I think this behavior is inconsistent. Both calls should raise the CS4014 if the returned `Task` is ignored silently.

**Comment by sharwell** [comment]
:arrow_right: Duplicate of #20782.
